applications/dashboard/settings/class.hooks.php

```diff
@@ -135,6 +135,14 @@ public function base_render_before($Sender) {
 
         // Allow global translation of TagHint
         $Sender->addDefinition("TagHint", t("TagHint", "Start to type..."));
+
+        // Add user's viewable roles to gdn.meta if user is logged in.
+        if (!array_key_exists('Roles', $Sender->getDefinitions())) {
```

Comment:
Some things I'm thinking about:

* `hasDefinition` is a really narrow use case.
* `addDefinition` already exists and, while terribly named, already does this technically.

So now I feel like we're fighting the framework.

Let's step back and rethink this. We're needing to detect this because we're hooking so late in the process, because we value allowing a plugin to override this. So it seems to me we should sidestep this entirely and hook earlier in the process so there's plenty of time left for a plugin to do otherwise. Let's hook into when the user's "session" is getting established and just `addDefinition` blindly like we do everywhere else.